man/man5/clients.5
.TH CLIENTS 5 "16 March 2001"
.SH NAME
clients \- RADIUS clients file
.SH DESCRIPTION
The \fIclients\fP file resides in the radius database directory,
by default \fI/etc/raddb\fP. Its use is depreciated in favour of 
\fIclients.conf\fP.

Every line starting with a hash sign
.RB (' # ')
is treated as comment and ignored.
.PP
Each line of the file contains two white-space delimited fields.
.IP client\ hostname
The RADIUS clients hostname.  This may be a plain hostname, or a
dotted-quad IP address.
.IP secret
This is the so-called "shared secret" that is held between a RADIUS
server and client. It is used to encrypt passwords in RADIUS packets,
and also for authentication. You need to configure the same secret
on the client (terminal server) as in this file.
.PP
The
.I clients
file is read by \fBradiusd\fP on startup only.
.SH FILES
.I /etc/raddb/clients
.SH "SEE ALSO"
.BR radiusd (8),
.BR clients.conf (5)
.BR naslist (5)


doc/ChangeLog
FreeRADIUS 1.1.0 ; $Date$, urgency=low
	Feature improvements
	* Modules can be load balanced, both in isolation and redundantly.
	  See doc/load-balance.txt for more information.
	* rlm_perl is now marked "stable"
	* N-tier certificate patch from Mohammed Petiwala.
	* Copied dictionaries from the CVS head (many, many, more vendors)
	* Enabled support for weird VSA formats, like Lucent and Starent.
	* Support encrypted IP address and integers, for Juniper clients.
	* Add PEAP machine authentication support in module "rlm_mschap".
	* Support User-Password field encryption in digest mode.
	* rlm_x99_token has become rlm_otp (with lots of changes).
	* Add rlm_sqlcounter to the list of stable modules.
	* Read MySQL specific options in sections [freeradius] and [client]
	  from file "my.cnf".
	* Support the ${Cisco-AVPair[n]} syntax.
	* Execute modules in {Pre,Post}-Proxy-Type stanzas.
	* Add new options to radclient to run stress tests on the server.
	* New module "rlm_sql_log" to postpone the storage of accounting data
	  in a SQL database. See rlm_sql_log(5) manpage.
	* New program "radsqlrelay" which sends the SQL logfile according to
	  the SQL server's capabilities.

	Bug fixes
	* rlm_digest has a number of bug fixes to authentication types.
	* Don't leak memory in module "rlm_sql".
	* Update the dictionaries, so that VALUEs with the same name,
	  but different numbers, aren't allowed.
	* Queue the request before looking for available threads.
	* Don't free the check items after we received the proxy reply.
	* Expand config variables in included files, too.
	* Check the return value of accounting modules and don't proxy
	  invalid requests.
	* In rlm_passwd, don't close a file stream more than once.
	* Fix format string errors in rlm_sql.c, spotted by Primoz Bratanic.
	* Walk the whole string in when escaping strings in rlm_ldap.
	* Include crypt.h if it is available so we get a prototype for crypt(),
	  spotted by Konstantin Kubatkin.
	* Removed (for almost all uses) length restrictions on vendor names
	  and VALUE names.
	* Don't leak memory when proxying an Access-Challenge response.
	* Make the sleep time user-defined, so radrelay can send more than
	  7 requests/s.
	* Fix a memory leak in rlm_checkval.
	* radclient doesn't resend countless times packets with invalid
	  signature.

FreeRADIUS 1.0.5 ; Date: 2005/09/04 16:23:00, urgency=medium
	Security Fixes
	* SQL injection attack in the module "rlm_sqlcounter".
	* Buffer overflows in the module "rlm_sqlcounter".
	* Expansion of variable %t may write 26 bytes beyond the buffer
	  bound. Primoz Bratanic is credited with the discovery of these
	  three bugs.

	Bug fixes
	* Don't de-reference a NULL pointer if the auth-type is unknown
	  in the function rad_check_password().
	* Escape more characters in the LDAP queries.
	  Bug found by Suse engineers.
	* In rlm_sql_unixodbc, don't call rad_malloc from sql_error(),
	  it leaks memory.
	* Fix an off-by-one error in the module rlm_sql_unixodbc.
	  Bug found by Suse engineers.
	* In rlm_sql, resize the buffer for the value of SQL-User-Name.
	* Initialize memory for a new SQL socket in the module rlm_sql.
	* Don't add too many attributes after running an external program.
	  Bug found by Suse engineers.
	* Fix an off-by-one error in the function getthing().
	* snprintf() and vsnprintf() replacements were not compiled if
	  the autoconf tests didn't find the functions.
	* Don't use vsprintf() anymore, but the replacement for vsnprintf()
	  in libradius instead.
	* The function decode_attribute() may write beyond buffer bounds.
	  Bug found by Suse engineers.
	* Fix a memset() in the function request_enqueue() which was
	  begining at the wrong address. Bug found by Matthias Ruttman.
	* Fix an off-by-one error in the function xlat_copy().
	  Bug found by Primoz Bratanic.
	* Fix other off-by-one errors in module "rlm_unix", too.
	  Bug found by Allan Bazinet.
	* Fix a 2-byte over-run read in function rad_decode().
	* Update thread pool queue properly.
	* Autonconf tests try first any user-specified directory,
	  otherwise they may pick up the wrong version.
	* Delete the autoconf tests for the libldap dependancies.
	* Install all the regular files under the "doc" directory.
	* Distinguish between exit code <0 (failure) and >0 (reject)
	  in Exec-Program-Wait. Patch from Thor Spruyt.
	* Make Expiration work.
	* Clean up the code for opening a proxy socket.
	* When finding a realm to proxy to, if all are dead, wake them
	  if wake_all_if_all_dead is true.
	* In radwho, print the NAS-Port as unsigned int.
	* Use extended regex instead of basic regex in rlm_attr_filter.
	* Catch the case where someone deletes a directory that rlm_detail
	  is using.
	* Use the variable $(LDFLAGS) when linking a module.
	* Ignore the Stripped-User-Name when a realm has the "nostrip"
	  directive.
	* Add support for NT-Password in rlm_pap.
	* In rlm_sqlcounter, use the time left to the next reset if it's
	  inferior to the time left in the counter.
	* Calculate Message-Authenticator correctly for Accounting-Request
	  and Accounting-Response.  Bug found by Paolo Rotela.
	* Build on MAC OS X.  Still need --disable-shared, though.
	* Fix bug #255 (crash with expired CRL's, etc.)
	* Fix quote removal of the values from a SQL database.
	* Reap the zombie process after a command run from "Exec-Program".
	* Allow to cancel proxy of accounting with "Proxy-To-Realm := LOCAL".
	* Don't copy VSA's to an Access-Reject packet.

FreeRADIUS 1.0.4 ; Date: 2005/06/11 22:46:52, urgency=medium
	* Fix installation problem.
	* Increase a buffer size, so radrelay doesn't truncate values.
	* Updates in the documentation. Patches from Thor Spruyt.

FreeRADIUS 1.0.3 ; Date: 2005/06/03 17:15:11, urgency=high
	Security Fixes
	* Always escape the strings in the SQL module.
	* Check buffer bound when input character needs escaping in
	  the SQL module. Bug found by Primoz Bratanic.

	Bug fixes

scripts/clients.pl
#!/usr/bin/env perl
#
#  Convert old-style "clients" file to new "clients.conf" format.
#
#  Usage: clients.pl clients [naslist] new-clients.conf
#	The "new-clients.conf" will be created if it does not exist.
#	If it does exist, it will be over-written.
#
#
#	$Id$
#
if (($#ARGV < 1) || ($#ARGV > 2)) {
    print "Usage: clients.pl clients [naslist] new-clients.conf\n";
    print "       The \"new-clients.conf\" will be created if it does not exist.\n";
    print "       If it does exist, it will be over-written.\n";
    exit(1);
}

$old = shift;
$new = shift;

if ($new =~ /naslist/) {
    $naslist = $new;
    $new = shift;
}

open OLD, "< $old" or die "Failed to open $old: $!\n";

while (<OLD>) {
    next if (/^\s*\#/);
    next if (/^\s*$/);

    split;

    $clients{$_[0]}{"secret"} = $_[1];
}
close OLD;

if (defined $naslist) {
    open OLD, "< $naslist" or die "Failed to open $naslist: $!\n";

    while (<OLD>) {
	next if (/^\s*\#/);
	next if (/^\s*$/);
	
	split;

	if (!defined $clients{$_[0]}) {
	    print "WARNING! client $_[0] is defined in naslist, but not in clients!";
	    next;
	}

	$clients{$_[0]}{"shortname"} = $_[1];
	$clients{$_[0]}{"nastype"} = $_[2];
    }
}

open NEW, "> $new" or die "Failed to open $new: $!\n";
foreach $client (keys %clients) {
    print NEW "client $client {\n";
    print NEW "\tsecret = ", $clients{$client}{"secret"}, "\n";
    if (defined $clients{$client}{"shortname"}) {
	print NEW "\tshortname = ", $clients{$client}{"shortname"}, "\n";
	print NEW "\tnastype = ", $clients{$client}{"nastype"}, "\n";
    }
    print NEW "}\n";
    print NEW "\n";
}
